pykmip_demo/README.md

```diff
@@ -1,15 +1,55 @@
+These instructions detail the install and setup processes for Accumulo, pelz, PyKMIP, and a proxy
+server. They proxy server coordinates communication between pelz and PyKMIP. This is because pelz
+does not support TLS endpoints within the Trusted-Exceuction Environment (TEE). The instructions
+then explain how to build Accumulo and run its unit tests using pelz and PyKMIP. The unit tests
+require successful functionality of the PelzCryptoService within Accumulo to protect the file-
+encryption-keys (FEKs) by using the key-encryption-key (KEK) stored within the PyKMIP server. 
 
-The demostration has four cert/key pairs and two keys for encryption. The two encryption keys are the File Encryption Key (FEK) and the Key Encryption Key (KEK). The KEK is keep on the PyKMIP server and will be securely transfered to the pelz enclave for FEK encyption.  The FEK will be generated by Accumulo for its internal file encryption then sent to pelz to be encrypted by the KEK for save storage of the FEK.  The four cert/key pairs are the one for the node, two for the proxy server, and one for PyKMIP server. The two proxy server cert/key pairs are for each individual connection to the node and PyKMIP.
+## Keys-in-use demonstration process
 
-The instructions assumes that SGX and TPM are already working on the demo system.
+The process for key protections is outlined here:  
+
+1. Accumulo generates a FEK for its internal file encryption
+2. Accumulo sends the FEK to pelz to be encrypted by the KEK. This request contains the KEK's
+  Unique Identification (UID)
+3. Pelz connects othe proxy server from within the TEE using an ECDH key establishment
+4. The proxy server forwards the key request to the PyKMIP server using TLS
+5. PyKMIP verifies the request and sends the KEK back to the pelz enclave through the proxy server
+6. Pelz uses the KEK within the enclave to encrypt/decrypt FEKs as per Accumulo's requests
+
+## Keys and Certificates
+
+The demostration has four cert/key pairs and two keys for encryption. The two encryption keys are:  
+
+ * The file-encryption-key (FEK)
+ * The key-encryption-key (KEK) 
+
+The KEK is stored on the PyKMIP server. It is used to wrap and unwrap FEKs as requested by Accumulo.
+The FEK is generated by Accumulo for its internal file protections.  
+
+The four certificate/key pairs are:  
+
+ * One certificate for the node
+ * Two for the proxy server
+ * One for PyKMIP server
+
+The proxy server has two certificate/key pairs to protect both the connection to the node and the
+connection to the PyKMIP server.
 
 ## Steps for End to End demo of PyKMIP, Accumulo, and pelz. 
+The remainder of this README describes the process for installing the necessary tools to execute
+the demonstration.
```

Comment:
I think we should add into this section some verbiage about the SGX setup steps are not included in the tool installation steps.